src/oemof/solph/custom.py

```diff
@@ -1567,9 +2210,2143 @@ def _objective_expression(self):
         for t in m.TIMESTEPS:
             for g in self.dsm:
                 dsm_cost += self.dsm_up[g, t] * g.cost_dsm_up
-                dsm_cost += sum(self.dsm_do[g, t, tt] for tt in m.TIMESTEPS
-                                ) * g.cost_dsm_down
+                dsm_cost += (sum(self.dsm_do_shift[g, t, tt]
+                                 for tt in m.TIMESTEPS) * g.cost_dsm_down_shift
+                             + self.dsm_do_shed[g, t] * g.cost_dsm_down_shed)
 
         self.cost = Expression(expr=dsm_cost)
 
         return self.cost
+
+
+class SinkDSMDIWInvestmentBlock(SinkDSMDIWBlock):
+    r"""Constraints for SinkDSM with "DIW" approach
+
+    **The following constraints are created for approach = 'DIW':**
+
+    .. _SinkDSMDelay-equations:
+
+    .. math::
+
+
+        &
+        (1) \quad \dot{E}_{t} = demand_{t} + DSM_{t}^{up} -
+        \sum_{tt=t-L}^{t+L} DSM_{t,tt}^{do}  \quad \forall t \in \mathbb{T} \\
+        &
+        (2) \quad DSM_{t}^{up} \cdot \eta = \sum_{tt=t-L}^{t+L} DSM_{t,tt}^{do}
+        \quad \forall t \in \mathbb{T} \\
+        &
+        (3) \quad DSM_{t}^{up} \leq  E_{t}^{up} \quad \forall t \in
+        \mathbb{T} \\
+        &
+        (4) \quad \sum_{t=tt-L}^{tt+L} DSM_{t,tt}^{do}  \leq E_{tt}^{do}
+        \quad \forall tt \in \mathbb{T} \\
+        &
+        (5) \quad DSM_{tt}^{up}  + \sum_{t=tt-L}^{tt+L} DSM_{t,tt}^{do} \leq
+        max \{ E_{tt}^{up}, E_{tt}^{do} \}\quad \forall tt \in \mathbb{T} \\
+        &
+        (6) \quad \sum_{tt=t}^{t+R-1} DSM_{tt}^{up} \leq E_{t}^{up} \cdot L
+        \quad \forall t \in \mathbb{T} \\
+        &
+
+
+
+    **Table: Symbols and attribute names of variables and parameters**
```

Comment:
This table reads almost the same as the previous one. Can you work with just one of the tables to avoid code duplication?

Reply:
Yes, there are quite some duplicates, thanks! I did the suggested quickfix, but will have a look on the tables and constraint formulations again, since I noticed that they need an update (I have not yet included all the adjustments I needed for investment modeling). But one table with all vars and params is a good suggestion.